test/unit/lib/plugins/aws/package/compile/layers/index.test.js

```diff
@@ -366,6 +366,16 @@ describe('lib/plugins/aws/package/compile/layers/index.test.js', () => {
       expect(layerResourceName).to.not.equal(layerResourceNamePrefix);
       const layerResource = cfResources[layerResourceName];
       expect(layerResource.DeletionPolicy).to.equal('Retain');
+
+      const layerPermissionResourceNamePrefix = naming.getLambdaLayerPermissionLogicalId(
+        'layerRetain'
+      );
+      const layerPermissionResourceName = Object.keys(cfResources).find(resourceName =>
+        resourceName.startsWith(layerPermissionResourceNamePrefix)
+      );
+      expect(layerPermissionResourceName).to.not.equal(layerPermissionResourceNamePrefix);
+      const layerPermissionResource = cfResources[layerPermissionResourceName];
+      expect(layerPermissionResource.DeletionPolicy).to.equal('Retain');
```

Comment:
Let's put it in another `it`, so it serves as individual spec item.

I'd probably name it: _should retain permission resource_